src/processor/result/factorized_table.cpp

```diff
@@ -292,6 +292,32 @@ void FactorizedTable::copyToInMemList(uint32_t colIdx, vector<uint64_t>& tupleId
     }
 }
 
+int64_t FactorizedTable::findFlatTupleIdxByColValue(uint64_t colIdx, int64_t relID) const {
+    auto tupleIdx = 0u;
+    auto numBlocks = numTuples / numTuplesPerBlock + 1;
```

Comment:
I don't think so. 
`numBlocks = numTuples / numTuplesPerBlock + 1;` is true only when all columns are flat. If ft has one unflat column, this may not necessarily be true. 
And i can't find other functions in FT who did the same caculation: `auto numBlocks = numTuples / numTuplesPerBlock + 1;`.
However, i think in this special case (where all columns are flat), i can directly get the numBlocks by: `auto numBlocks = flatTupleBlockCollection->getNumBlocks();`